perf/perf_24x7_all_events.py

```diff
@@ -0,0 +1,145 @@
+        if 'ppc' not in process.system_output("uname -p", ignore_status=True):
+            self.cancel("Processor is not ppc64")
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in detected_distro.name:
+            deps.extend(['linux-tools-common', 'linux-tools-%s'
+                         % platform.uname()[2]])
+        elif detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(['perf', 'numactl'])
+        else:
+            self.cancel("Install the package for perf supported by %s"
+                        % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        self.cpu_arch = cpu.get_cpu_arch().lower()
+        self.perf_args = "perf stat -v -C 0 -e"
+        if self.cpu_arch == 'power8':
+            self.perf_stat = "%s hv_24x7/HPM_0THRD_NON_IDLE_CCYC" % self.perf_args
+        if self.cpu_arch == 'power9':
+            self.perf_stat = "%s hv_24x7/CPM_TLBIE" % self.perf_args
+        self.event_sysfs = "/sys/bus/event_source/devices/hv_24x7"
+
+        # Check if this is a guest
+        # 24x7 is not suported on guest
+        cpu_output = process.run("cat /proc/cpuinfo")
+        if "emulated by" in cpu_output.stdout:
+            self.cancel("This test is not supported on guest")
+
+        # Check if 24x7 is present
+        if os.path.exists("%s" % self.event_sysfs):
+            self.log.info('hv_24x7 present')
+        else:
+            self.cancel("%s doesn't exist.This feature is supported"
+                        " only on LPAR" % self.event_sysfs)
+
+        # Performance measurement has to be enabled in lpar through BMC
+        # Check if its enabled
+        result_perf = process.run("%s,domain=2,core=1/ sleep 1"
+                                  % self.perf_stat, ignore_status=True)
+        if "not supported" in result_perf.stderr:
+            self.cancel("Please enable LPAR to allow collecting"
+                        " the 24x7 counters info")
+
+        # Create the file with contents of hv_24x7 events
+        self.perf_list_file = tempfile.NamedTemporaryFile().name
+        output = process.run("perf list")
+        for line in output.stdout.split('\n'):
+            if 'hv_24x7' in line:
+                line = line.split(',')[0].split('/')[1]
+                genio.append_one_line(self.perf_list_file, line)
+
+        # Getting the number of cores
+        output = process.run("lscpu")
+        for line in output.stdout.split('\n'):
+            if 'Core(s) per socket:' in line:
+                self.cores = int(line.split(':')[1].strip())
+
+        # Getting the number of chips available in the machine
+        output = process.run('numactl -H')
+        for line in output.stdout.split('\n'):
+            if 'available' in line:
+                self.chip = int(line.split('-')[1].split(')')[0].strip())
```

Comment:
This will fail on machines without range. ex. `available: 2 nodes (0,8)`

Reply:
@harish-24 In some of the machines I can see `available: 2 nodes (0-1)`. As you pointed out, it might be an issue with the non-continuous nodes. I will have a look. Thanks!!